### PR TITLE
Edit the methods used in the acceptance tests to filter the resources using keyword

### LINKED ISSUE #2679: Add acceptance test to filter files, folders, and hidden elements using keyword in the shared-with-me page
Related to issue #2482 
- [x] -Add new tests to filter files, folders and hidden elements using keyword in the shared-with-me page

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Edit the methods used in the acceptance tests to filter the resources using keyword

## Related Issue

- Fixes  #2679

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RenameFolders</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7184/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7184/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>RenameFolders</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7189/

<p>






</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7195/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7195/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7199/

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>iPhone</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7207/

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>XGAPortrait</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/7207/

<p>





</p></details>

File: tests/acceptance/features/webUITrashbin/trashbinFilter.feature [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      | .hiddenTestFile     |
+      | testFolder          |
+    And the user has browsed to the trashbin page
+
+  Scenario: user filters files and folders using keyword when the files filter, folders filter, and hidden filter are enabled
+    When the user enables file filter using the webUI
+    And the user enables folder filter using the webUI
+    And the user enables the setting to view hidden files on the webUI
+    And the user filters the file list by "simple" on the webUI
+    Then all files and folders containing pattern "simple" in their name should be listed in the trash on the webUI
```
Comment (dpakach) [inline tests/acceptance/features/webUITrashbin/trashbinFilter.feature]:
```suggestion
    Then all the trashbin files and folders containing pattern "simple" in their name should be listed in webUI
```
This sounds better IMO

File: tests/acceptance/features/webUISharingNotifications/sharedWithMeFilter.feature [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  As a user
+  I would like to filter files/folders
+  So that I can make the file/folder list more clear
+
+  Background:
+    Given these users have been created with default attributes:
+      | username |
+      | user1    |
+      | user2    |
+    And user "user1" has created file ".simpleHiddenFile"
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/acceptance/features/webUISharingNotifications/sharedWithMeFilter.feature]: in next pr

File: tests/acceptance/features/webUITrashbin/trashbinFilter.feature (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/features/webUITrashbin/trashbinFilter.feature]: in next pr

**Comment by individual-it** [comment]
this changes are also contained in  #2708 closing here?

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 3]
```diff
@@ -18,7 +18,9 @@ const { getSharedWithOthersFiles, getSharedWithOthersFolders } = require('../hel
 
 const assertDesiredResourcesListed = async function (resourcesMatchingPattern, listedResources) {
   const diff = _.difference(resourcesMatchingPattern, listedResources)
-  return assert.strictEqual(diff.length, 0, `Expected : ${resourcesMatchingPattern} but got ${listedResources}`)
+  const difference = _.difference(listedResources, resourcesMatchingPattern)
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: give it a better name, to confusing with `diff`

Comment (jasson99) [reply tests/acceptance/stepDefinitions/filesContext.js]: fixed

**Pull request merged**